src/realm.h

```diff
@@ -1060,6 +1064,30 @@ RLM_API bool realm_commit(realm_t*);
  */
 RLM_API bool realm_rollback(realm_t*);
 
+/**
+ * start a new write transaction asynchronously for the realm passed as argument.
+ */
+RLM_API realm_async_transaction_token_t* realm_async_begin_transaction(realm_t* realm,
+                                                                       realm_on_realm_async_begin_transaction_func_t,
+                                                                       realm_userdata_t userdata,
+                                                                       realm_free_userdata_func_t userdata_free,
+                                                                       bool* notify_only);
```

Comment:
I would make this a plain boolean, not a pointer to an optional value - setting this up from something like Dart would be too much of a nuisance, and SDKs always passing in the default value is faster.